utilities/src/main/java/ru/study21/jcsv/xxl/analyzer/CSVCustomizableAnalyzer.java

```diff
@@ -146,15 +162,88 @@ public void acceptRow(List<String> row) {
                     maxValuesIntAction(colIndex, 1);
 
             @Override
-            public void acceptRow(List<String> row) {
+            public void acceptRow(CSVRow row) {
                 delegate.acceptRow(row);
             }
 
             @Override
             public Integer getResult() {
                 return delegate.getResult().get(0);
             }
+
+            @Override
+            public void finish() {}
+        };
+    }
+
+    public static Action<Long> withOffsetsSizeAction(int colIndex) {
+        return new Action<>() {
+            Long size = 0L;
+
+            @Override
+            public void acceptRow(CSVRow row) {
+                size += Long.SIZE + row.get(colIndex).getBytes(StandardCharsets.UTF_8).length;
+            }
+
+            @Override
+            public Long getResult() {
+                return size;
+            }
+
+            @Override
+            public void finish() {}
+        };
+    }
+
+    public static Action<Boolean> offsetsWriteAction(BufferedWriter writer, int colIndex) {
+        return new Action<>() {
+            boolean result = true;
+
+            @Override
+            public void acceptRow(CSVRow row) {
+                try {
+                    writer.write(String.join(",", String.valueOf(row.offset()), row.get(colIndex)) + "\n");
+                } catch (IOException e) {
```

Comment:
First of all, we keep data for huge files. And data could be huge.
So it makes sense to keep them compact.

And what we need is pair of row number and its byte offset in file (see another my long comment).

And as we need offset info for each row and we are going to ask offset of row (not opposite) we can:

1. Save byte offset in binary form as 8-byte values (I guess it is absolutely safe to assime that 8 bytes is enough for size)
2. Not save row number at all. Consider it as kind of index in big array of byte offsets.

We also need byte size of each row but this info is implicitly available as diff of offsets of current and next row.

(Indeed it would be more compact to store just offsets but it is not so good indeed because we'll need to know byte offsets in random order so this is not an option for us)